examples/common/src/chariott/streaming.rs

```diff
@@ -111,22 +66,9 @@ where
     }
 }
 
-#[async_trait]
-impl<T> ChannelService for StreamingStore<T>
-where
-    T: Clone + Send + Sync + 'static,
-{
-    type OpenStream = ReceiverStream<Result<ProtoEvent, Status>>;
-
-    async fn open(
-        &self,
-        _: tonic::Request<OpenRequest>,
-    ) -> Result<Response<Self::OpenStream>, Status> {
-        let id: ClientId = uuid::Uuid::new_v4().to_string().into();
-        let (_, receiver_stream) = self.ess.read_events(id.clone());
-        let mut response = Response::new(receiver_stream);
-        response.metadata_mut().insert(METADATA_KEY, id.to_string().try_into().unwrap());
-        Ok(response)
+impl<T> AsRef<InnerEss<(EventId, T)>> for StreamingStore<T> {
```

Comment:
I don't think a `StreamingStore<T>` should _pretend_ to be an ESS. I would just provide a simple getter that's clear:

```rust
impl<T> StreamingStore<T> {
    pub fn ess(&self) -> &InnerEss<(EventId, T)> {
        &self.ess.0
    }
}
```

Reply:
Thanks for the suggestion, done with 1ce2e6ef20a5b9e76f41fc82675bcf9fcab5423a.